src/plugins/flow.js

```diff
@@ -1162,6 +1162,7 @@ export default function (instance) {
         let state = this.state.clone();
         try {
           let returnType = this.flowParseTypeAnnotation();
+          if (this.canInsertSemicolon()) this.unexpected();
```

Comment:
Not entirely sure if that has any side effects on this try/catch logic.
So maybe we jsut leave it for now, and later on have a look how we can improve the message.